docs/source/x_ext.rst

```diff
@@ -366,24 +394,24 @@ Compressed interface
   :class: no-scrollbar-table
 
-  +------------------------+-------------------------+-----------------------------------------------------------------------------------------------------------------+
-  | **Signal**             | **Type**                | **Description**                                                                                                 |
-  +------------------------+-------------------------+-----------------------------------------------------------------------------------------------------------------+
-  | ``instr``              | logic [15:0]            | Offloaded compressed instruction.                                                                               |
-  +------------------------+-------------------------+-----------------------------------------------------------------------------------------------------------------+
-  | ``id``                 | :ref:`id_t <id>`        | Identification number of the offloaded compressed instruction.                                                  |
-  +------------------------+-------------------------+-----------------------------------------------------------------------------------------------------------------+
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | **Signal**             | **Type**                 | **Description**                                                                                                 |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``instr``              | logic [15:0]             | Offloaded compressed instruction.                                                                               |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
+  | ``hartid``             | :ref:`hartid_t <hartid>` | Identification of the hart offloading the instruction.                                                          |
+  +------------------------+--------------------------+-----------------------------------------------------------------------------------------------------------------+
 
 The ``instr[15:0]`` signal is used to signal compressed instructions that are considered illegal by |processor| itself. A |coprocessor| can provide an uncompressed instruction
 in response to receiving this.
 
-A compressed request transaction is defined as the combination of all ``compressed_req`` signals during which ``compressed_valid`` is 1 and the ``id`` remains unchanged.
+A compressed request transaction is defined as the combination of all ``compressed_req`` signals during which ``compressed_valid`` is 1 and the ``hartid`` remains unchanged.
```

Comment:
All places where the "hartid remains unchanged" is used should be replaced by the "hartid and id remain unchanged". The reason is that this aids in implementing an XIF-based interconnect. The idea is that the compressed/issue interface can be broadcast to all coprocessors and if one of them signals a ready the hartid and/or id might progress to new values before all coprocessors actually signalled a ready.

Reply:
I can agree to the use-case in general. Although it seems unlikely that decode of an offloaded instruction would take longer than one cycle. I cannot make the change as you suggest, as I did replace the ID signal in the compressed interface with hartid. It is a separate commit, so it is easily reverted. I don't have strong opinions on the compressed interface. If you would like that reverted. I think of the compressed interface as a simple combinatorial interface, and that would not need an ID. Leaving the decision to you

Reply:
I am fine with having no ID for the compressed interface, but all places in this spec that still refer to hartid remaining unchanged will need to be updated

Reply:
fixed in 08f84d1, commits squashed for merge into main